Component/MvuDrawingApp/App.fs

```diff
@@ -0,0 +1,242 @@
+        )
+            .dock(Dock.Bottom)
+            .margin(5.0)
+            .padding(5.0)
+            .cornerRadius(8.0)
+            .background("#bdc3c7")
+
+module DrawingCanvas =
+    type Model =
+        { IsPressed: bool
+          LastPoint: Point option
+          Color: Color
+          Size: float }
+
+    type Msg =
+        | PointerPressed of Input.PointerPressedEventArgs
+        | PointerReleased of Input.PointerReleasedEventArgs
+        | PointerMoved of Input.PointerEventArgs
+
+    let init () =
+        { IsPressed = false
+          LastPoint = None
+          Color = Colors.Black
+          Size = 2. }
+
+    let canvasRef = ViewRef<Canvas>()
+
+    let update msg model =
+        match msg with
+        | PointerPressed _ -> { model with IsPressed = true }
+        | PointerReleased _ -> { model with IsPressed = false }
+        | PointerMoved args ->
+            let point = args.GetPosition(canvasRef.Value)
+
+            if model.IsPressed then
+                match model.LastPoint with
+                | Some lastPoint ->
+                    let brush = unbox(ColorToBrushConverter.Convert(box(model.Color), typeof<IBrush>))
+
+                    let line =
+                        Shapes.Line(StartPoint = lastPoint, EndPoint = point, Stroke = brush, StrokeThickness = model.Size, StrokeLineCap = PenLineCap.Round)
+
+                    if canvasRef.Value <> null then
+                        canvasRef.Value.Children.Add(line)
+
+                    { model with LastPoint = Some point }
+                | None -> { model with LastPoint = Some point }
+            else
+                { model with LastPoint = Some point }
+
+    let view () =
+        Canvas(canvasRef)
+            .verticalAlignment(VerticalAlignment.Stretch)
+            .horizontalAlignment(HorizontalAlignment.Stretch)
+            .background(SolidColorBrush(Colors.White))
+            .onPointerPressed(PointerPressed)
+            .onPointerReleased(PointerReleased)
+            .onPointerMoved(PointerMoved)
+
+
+module App =
+    let theme = FluentTheme()
+
+    type Model =
+        { Setting: Setting.Model
+          DrawingCanvas: DrawingCanvas.Model }
+
+    type Msg =
+        | SettingMsg of Setting.Msg
+        | DrawingCanvasMsg of DrawingCanvas.Msg
+
+    let init () =
+        { Setting = Setting.initModel
+          DrawingCanvas = DrawingCanvas.init() }
+
+    let update msg model =
+        match msg with
+        | SettingMsg msg ->
+            let setting = Setting.update msg model.Setting
+            { model with Setting = setting }
+
+        | DrawingCanvasMsg msg ->
+            let drawingCanvas =
+                { model.DrawingCanvas with
+                    Color = model.Setting.ColorPicker.Color
+                    Size = model.Setting.SizePicker.Size }
+
+            let drawingCanvas = DrawingCanvas.update msg drawingCanvas
+
+            { model with
+                DrawingCanvas = drawingCanvas }
```

Comment:
@TimLariviere I was interested in how to get this data after we removed the `View.map` as this not longer will be called right ?

Reply:
In which case, you need to use Binding.

The parent component will host the state that child components can then listen and update.

```fs
let app () =
    Component() {
        let! color = Context.State(Color.Red)

        VStack() {
            setting color
            drawingCanvas color
        }
    }

let setting (sharedColor: State<Color>) =
    Component(settingProgram) {
        let! color = Context.Binding(sharedColor)
    
        // Updating the binding here will trigger the canvas to refresh
        Button("Set color to blue", fun () -> color.Set(Color.Blue))
    }

let drawingCanvas (sharedColor: State<Color>) =
    Component(drawingProgram) {
        let! color = Context.Binding(sharedColor)

        Canvas()
            .fill(color.Current)
    }
```